### PR TITLE
Add user connections history component and enhance translations

### PR DESCRIPTION
**Pull request by stephdl**
Introduce a new component for displaying user connections history in OpenVPN and improve translation strings to support this feature.

### CODE REVIEW

File: src/components/standalone/openvpn_rw/ConnectionsHistory.vue [hunk 1]
```diff
@@ -0,0 +1,347 @@
...
+    const matchesObjectTypeFilter =
+      objectAccountsFilter.length === 0 || objectAccountsFilter.includes(ConnectionsRecord.account)
+
+    return matchesFilter && matchesTimeRange && matchesObjectTypeFilter
+  })
+}
+
+// filter items
+const filteredItems = computed(() => {
+  if (timeRangeFilter.value[0] === 'all') {
```
[the remark this answers is not in the document]
Comment (stephdl) [reply src/components/standalone/openvpn_rw/ConnectionsHistory.vue:148]: your solution output always all entries, but anyway with includes it is better, nice fix

// filter items
const filteredItems = computed(() => {
  if (['all', 'today', 'last_week', 'last_month', 'last_3_months'].includes(timeRangeFilter.value[0])) {
    return applyFilterToConntrackRecords(
      connectionsRecords.value,
      textFilter.value,
      timeRangeFilter.value[0],
      accountsFilter.value
    )
  } else {
    return connectionsRecords.value
  }
})

[the remark this answers is not in the document]
Comment (andre8244) [reply src/components/standalone/openvpn_rw/ConnectionsHistory.vue:148]: Yeah sorry, `<item> in <list>` is Python :man_facepalming:

File: src/components/standalone/openvpn_rw/UserConnectionsData.vue [hunk 2]
```diff
@@ -0,0 +1,200 @@
...
+  sortKey.value = payload.key
+  sortDescending.value = payload.descending
+}
+
+const { currentPage, paginatedItems } = useItemPagination(() => sortedItems.value, {
+  itemsPerPage: pageSize
+})
+
+// Format duration in seconds to human readable format
+function formatDuration(seconds: number): string {
```
[the remark this answers is not in the document]
Comment (stephdl) [reply src/components/standalone/openvpn_rw/UserConnectionsData.vue:81]: yep lucia complained that she wanted something short

File: src/components/standalone/openvpn_rw/UserConnectionsData.vue [hunk 3]
```diff
@@ -0,0 +1,200 @@
...
+const sortKey = ref('startTime')
+const sortDescending = ref(true)
+
+// Convert IP address to integer for sorting
+function ipToInt(ip: string): number {
+  return ip.split('.').reduce((acc, octet) => (acc << 8) + parseInt(octet, 10), 0)
+}
+
+// Custom sorting functions
+function sortFunctions(a: ConnectionsRecord, b: ConnectionsRecord) {
```
[the remark this answers is not in the document]
Comment (stephdl) [reply src/components/standalone/openvpn_rw/UserConnectionsData.vue]: done

File: src/components/standalone/openvpn_rw/UserConnectionsTable.vue [hunk 4]
```diff
@@ -0,0 +1,212 @@
...
+    if (octets1[i] !== octets2[i]) {
+      return octets1[i] - octets2[i]
+    }
+  }
+  return 0
+}
+
+// Custom sorting functions
+const sortFunctions = {
+  account: (a: ConnectionsRecord, b: ConnectionsRecord) => {
```
Comment (andre8244) [inline src/components/standalone/openvpn_rw/UserConnectionsTable.vue:50]: Inside `sortFunctions` we need to specify only custom sorting functions: default sorting functions are already implemented in [useSort.ts composable](https://github.com/nethesis/vue-components/blob/main/src/composables/useSort.ts#L14).

Therefore, `account` key and function can be removed, and probably also `startTime`, `endTime`, `duration`, `bytesReceived` and `bytesSent`. You can try to remove them and check if the table sorting still works.

On the other hand, `virtualIpAddress` and `remoteIpAddress` are definitely needed here.

Comment (stephdl) [reply src/components/standalone/openvpn_rw/UserConnectionsTable.vue:50]: image
image
if you do not mind I prefer to keep my function for endTime and duration it seems that the missing value replaced by `-` makes some strange behaviour with the default sorting function

Comment (andre8244) [reply src/components/standalone/openvpn_rw/UserConnectionsTable.vue]: If the strange behavior is that rows without a value appear first or last, that's perfectly normal. Anyway non problem, you can leave it as it is now.

**Review by andre8244 — APPROVED** [review]
LGTM, you can squash & merge

**Comment by Tbaile** [comment]
For next time please `squash and merge` formatting the title with a conventional commit standard: https://www.conventionalcommits.org/en/v1.0.0/

I'd suggest naming the PR when opening it, since the squash will retain such title as a commit message

Reference of the tool used to release: https://github.com/googleapis/release-please

**Pull request merged**